### PR TITLE
DROOLS-1692: Align optaplanner-peristence-jpa with Narayana

### PR DESCRIPTION
**Pull request by domhanak**
Replace bitronix with narayana
Introduce new class contaning properties for optaplanner
persistence tests.

Depends on #https://github.com/kiegroup/drools/pull/1455

@baldimir

### CODE REVIEW

File: optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <persistence xmlns="http://java.sun.com/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              version="2.0"  xsi:schemaLocation="http://java.sun.com/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_0.xsd">
-  <persistence-unit name="optaplanner-persistence-jpa-test" transaction-type="JTA">
-    <provider>org.hibernate.ejb.HibernatePersistence</provider>
+  <persistence-unit name="org.optaplanner.persistence.jpa" transaction-type="JTA">
```
Comment (ge0ffrey) [inline optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml]: we'll want to keep the word test in here as it's a test

File: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java [hunk 2]
```diff
@@ -46,9 +48,9 @@
 
     @Before
     public void setUp() throws Exception {
-        context = PersistenceUtil.setupWithPoolingDataSource("optaplanner-persistence-jpa-test");
+        context = OptaplannerPersistenceUtil.setupWithPoolingDataSource(OptaplannerPersistenceUtil.OPTAPLANNER_PERSISTENCE_UNIT_NAME);
```
Comment (ge0ffrey) [inline optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java]: Extracting this lookup name to a separate, new class is just verbosity as it's only used once.
We can always do taht later if it's ever used twice (which won't be the case).

File: optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+    <provider>org.hibernate.jpa.HibernatePersistenceProvider</provider>
```
Comment (ge0ffrey) [inline optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml]: 1) For my info, what's the difference between the ejb and jpa hibernate persistence provider?
2) Why do we configure one to begin with? It is related to the database sql dialect detection?

Comment (baldimir) [reply optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml]: Oh I missed this. org.hibernate.ejb.HibernateProvider is deprecated. @domhanak why was this change needed? Or is it just a change copied from jBPM changes? Please try if the changes work leaving the original provider there.

Comment (domhanak) [reply optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml]: @ge0ffrey 
1. From what I found - One is Hibernate persistence provider implementation and the other is Hibernate EJB3 persistence provider implementation.

@baldimir 
Reason i changed this is that org.hibernate.ejb.HibernatePersistence is deprecated  - https://docs.jboss.org/hibernate/orm/5.1/javadocs/org/hibernate/ejb/HibernatePersistence.html
Furthermore, org.hibernate.ejb.HibernateProvider is not part of newer versions of hibernate.
https://docs.jboss.org/hibernate/orm/5.2/javadocs/org/hibernate/ejb/HibernatePersistence.html

2. It is invoked by the Persistence class in Java SE environments to create an EntityManagerFactory. SQL Dialect is handled by dialect classes and is DB specific.

File: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java (hunk 2 above)
Comment (domhanak) [reply optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java]: Agreed, will change it so there is no new class.

File: optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml (hunk 1 above)
Comment (domhanak) [reply optaplanner-persistence/optaplanner-persistence-jpa/src/test/filtered-resources/META-INF/persistence.xml]: +1

**Comment by yurloc** [comment]
Please fix the typo in commit message (allign →align).

**Comment by domhanak** [comment]
@ge0ffrey Incorporated the changes you requested and corrected typo in commit message

**Comment by ge0ffrey** [comment]
Thanks @domhanak et al :)

**Pull request merged**